frontend/src/dashboard/DashboardPageController.ts

```diff
@@ -1,10 +1,66 @@
+import OneTimeTransaction from "../common/Classes/OneTimeTransaction";
+import Transaction from "../common/Classes/Transaction";
 import User from "../common/Classes/User";
 import PageController from "../common/Interfaces/PageController";
 
 export default class DashboardPageController implements PageController {
+  private amountSpentSpan: HTMLElement
+  private transactionsTable: HTMLTableElement
+  private user: User
+  private userTransactions: Array<Transaction>
 
-    public run(user: User): void {
-        console.log("Running DashboardPageController");
-    }
+  public run(user: User): void {
+    this.user = user
     
-}
\ No newline at end of file
+    this.transactionsTable = <HTMLTableElement>document.getElementById("trans-table")
+    this.amountSpentSpan = document.getElementById("amount");
+    this.createDummyPayments()
+
+    this.userTransactions = this.user.getAllTransactions()
+    const totalSpent =this.userTransactions.reduce((prev, current) => {
+      return prev + current.getAmount()

```

Comment:
Not all transactions are of type "expense". Check the type of the transaction before adding it to `totalSpent`